src/main/webapp/go.js

```diff
@@ -150,30 +185,38 @@ function toggleHintButton(hide) { //eslint-disable-line
  * Disable lint check because getHint() is called from go.html.
  */
 function getHint() { //eslint-disable-line
-  if (destIndex >= 0) {
-    const hintArr = huntArr[destIndex].hints;
-    if (hintIndex < hintArr.length) {
-      updateMessage(HINT_DISPLAY, hintArr[hintIndex]);
-      hintIndex++;
+  if (hunt.getDestIndex() >= 0) {
+    const hintArr = hunt.getCurHints();
+    if (hunt.getHintIndex() < hintArr.length) {
+      updateMessage(HINT_DISPLAY, hintArr[hunt.getHintIndex()]);
+      hunt.setHintIndex(hunt.getHintIndex() + 1);
```

Comment:
Might be more cohesive to set hunt.getHintIndex() as another const variable, since you're getting it three times